### DISCUSSION TITLE
Figure BV for sulcal morphometry

### DISCUSSION THREAD

**Discussion by andreaPay** [brainvisa/brainvisa.github.io#182] [Q&A]
Hello everyone, 

I would like to do a figure displaying p values or T with a colormap on sulci such as in Duchesnay et al. I3ETMI, 2007 figure 8. 
Could you help me to do this ? 

Thanks a lot !

**Comment by denisri** [comment]
Hi,
Well, we used to have a command for that, but we have not used it for a while and I realize that it is broken because we have not updated it after some libraries refactoring. We have to fix it.
In the meantime, you can use a python script like this one (to be adapted):
```
import anatomist.api as ana
from soma.qt_gui.qt_backend import Qt
import csv


with open('/tmp/sulci_values.csv') as f:
    r = csv.reader(f)
    table = {line[0].strip(): [float(x.strip()) for x in line[1:]]
             for line in r}

a = ana.Anatomist()
graph = a.loadObject('Lsujet01_default_session_auto.arg')

ag = graph.toAimsObject()

for v in ag.vertices():
    label = v.get('label')
    value = table.get(label)
    if value is not None:
        v['value'] = value[0]

mesh = a.loadObject('sujet01_Lwhite.gii')

w = a.createWindow('3D')
w.addObjects([graph, mesh])
graph.setColorMode(graph.PropertyMap)
graph.setColorProperty('value')
graph.setPalette('Blue-Red')

qapp = Qt.QApplication.instance()
qapp.exec()
```
with a CSV file (`sulci_values.csv`) looking like this:
```
S.C._left,           10.5
S.Pe.C.sup._left,    3.
S.Pe.C.inf._left,    5.
```

You should run the script inside the BrainVisa container environment, ie if the script is saved as `sulcicsv.py`, then run:
```
bv sulcicsv.py
```

Denis

**Reply by andreaPay** [reply]
Thank you a lot ! I am ging to try ! 

Best regards, 

Andréa
